phygnn/model_interfaces/base_model.py

```diff
@@ -290,6 +293,39 @@ def label_stdevs(self):
 
         return stdevs
 
+    @property
+    def one_hot_input_feature_names(self):
+        """
+        Input feature names to be one-hot encoded
+
+        Return
+        ------
+        list
+        """
+        return list(self._one_hot_categories.keys())
+
+    @property
+    def one_hot_feature_names(self):
+        """
+        One-hot encrypted feature names
+
+        Return
+        ------
+        list
+        """
+        return [i for l in self._one_hot_categories.values() for i in l]
```

Comment:
We should just add a check in the preprocessing with this to make sure all of these values are unique and that none overlap with the non-OHE feature names.

Reply:
Agreed, thoughts on where to put this:
1) pre-flight check in the init?
2) In make_one_hot_feature_names?

Reply:
I think we should pass in the column labels of `self._features` to `_make_df_one_hot_cols_labels()` and have that method check that the new one hot cols labels are "legal". That would work right? The model interfaces pass in dataframes to the one hot utility? Obviously if the one hot utility gets array data it will not run this check. 

https://github.com/NREL/phygnn/blob/3345f50aca33521e893abf587ff62a90a924223b/phygnn/utilities/pre_processing.py#L332-L335

https://github.com/NREL/phygnn/blob/3345f50aca33521e893abf587ff62a90a924223b/phygnn/utilities/pre_processing.py#L226

Reply:
Added the this to the check_one_hot_categories method (see above). I run this in two places:
1) in the init to check for duplicate categories
2) in make_one_hot_feature_names to check for overlap with the feature_names